src/python.cpp

```diff
@@ -1467,7 +1467,7 @@ SEXP py_get_formals(PyObjectRef callable)
     const char *name_char = PyUnicode_AsUTF8(name);
     if (name_char == NULL) throw PythonException(py_fetch_error());
 
-    SEXP name_sym = Rf_installTrChar(Rf_mkCharCE(name_char, CE_UTF8));
+    SEXP name_sym = Rf_installChar(Rf_mkCharCE(name_char, CE_UTF8));
```

Comment:
I _think_ you need to explicitly translate the CHARSXP here from UTF-8 to the native encoding, at least for older versions of R?

Reply:
Yeah, `installChar()` in older versions of R assumes that the CHARSXP is already in the native encoding, so I think you should translate first.

Reply:
I think `Rf_installChar()` just calls `Rf_installTrChar()`, at least in R 3.5 and 4.3 ( 1st and 3rd link in https://github.com/rstudio/reticulate/issues/1388#issuecomment-1581245580)

Reply:
i.e., `installChar()` and `installTrChar()` are functionally identical.

Reply:
This is the definition I see in the R 3.4.4 sources:

```
/* This function is equivalent to install(CHAR(charSXP)), but faster.
   Like the equivalent code pattern, it discards the encoding information,
   hence in almost all cases installTrChar should be used, instead. */
SEXP installChar(SEXP charSXP)
{
    SEXP sym;
    int i, hashcode;

    if( !HASHASH(charSXP) ) {
	hashcode = R_Newhashpjw(CHAR(charSXP));
	SET_HASHVALUE(charSXP, hashcode);
	SET_HASHASH(charSXP, 1);
    } else {
	hashcode = HASHVALUE(charSXP);
    }
    i = hashcode % HSIZE;
    /* Check to see if the symbol is already present;  if it is, return it. */
    for (sym = R_SymbolTable[i]; sym != R_NilValue; sym = CDR(sym))
	if (strcmp(CHAR(charSXP), CHAR(PRINTNAME(CAR(sym)))) == 0) return (CAR(sym));
    /* Create a new symbol node and link it into the table. */
    int len = LENGTH(charSXP);
    if (len == 0)
	error(_("attempt to use zero-length variable name"));
    if (len > MAXIDSIZE)
	error(_("variable names are limited to %d bytes"), MAXIDSIZE);
    if (IS_ASCII(charSXP) || (IS_UTF8(charSXP) && utf8locale) ||
					(IS_LATIN1(charSXP) && latin1locale) )
	sym = mkSYMSXP(charSXP, R_UnboundValue);
    else {
	/* This branch is to match behaviour of install (which is older):
	   symbol C-string names are always interpreted as if
	   in the native locale, even when they are not in the native locale */
	PROTECT(charSXP);
	sym = mkSYMSXP(mkChar(CHAR(charSXP)), R_UnboundValue);
	SET_HASHVALUE(PRINTNAME(sym), hashcode);
	SET_HASHASH(PRINTNAME(sym), 1);
	UNPROTECT(1);
    }

    R_SymbolTable[i] = CONS(sym, R_SymbolTable[i]);
    return (sym);
}
```

If we want to support R 3.4.x, then I think we do need to translate first.